src/dx_device_twins.c

```diff
@@ -220,6 +223,18 @@ static void SetDesiredState(JSON_Object *jsonObject, DX_DEVICE_TWIN_BINDING *dev
             deviceTwinBinding->propertyValue = NULL;
         }
         break;
+    case DX_DEVICE_TWIN_JSON_OBJECT:
+        if (json_object_has_value_of_type(jsonObject, deviceTwinBinding->propertyName,
+                                          JSONObject)) {
+            deviceTwinBinding->propertyValue =
+                (JSON_Object *)json_object_get_object(jsonObject, deviceTwinBinding->propertyName);
+
```

Comment:
thoughts on including the following line (same question for DX_DEVICE_TWIN_STRING case):

deviceTwinBinding->propertyUpdated = true;

Reply:
String and JsonObject types are special cases compared to the other scalar types.  In the scaler type cases memory is allocated (see deviceTwinOpen()) and the device twin value is captured in that memory, then the propertyUpdated flag set to true.  Since the String and JsonObject cases don't allocate memory to hold the value part (the sizes are unknown until the data arrives), that flag is not utilized.